components/script/dom/performance.rs

```diff
@@ -273,10 +287,70 @@ impl Performance {
             o.notify();
         }
     }
-
     fn now(&self) -> f64 {
         (time::precise_time_ns() - self.navigation_start_precise).to_ms()
     }
+    fn can_add_resource_timing_entry(&self) -> bool {
+        self.resource_timing_buffer_current_size.get() <=
+            self.resource_timing_buffer_size_limit.get()
+    }
+    fn copy_secondary_resource_timing_buffer(&self) {
+        while self.can_add_resource_timing_entry() && !self
+            .resource_timing_secondary_entries
+            .borrow_mut()
+            .is_empty()
+        {
+            self.queue_entry(
+                &(*(self
+                    .resource_timing_secondary_entries
+                    .borrow_mut()
+                    .pop_front()
+                    .unwrap())),
+                true,
+            );
+        }
+    }
+    fn fire_buffer_full_event(&self) {
```

Comment:
nit: one thing that helps reading the code and comparing it with the spec is adding the link to the specified algorithm above the function definition and a comment identifying the different steps. For example:
```rust
// `fire a buffer full event` paragraph of
// https://w3c.github.io/resource-timing/#sec-extensions-performance-interface
fn fire_buffer_full_event(&self) {
    // Step 1.
            while !self
            .resource_timing_secondary_entries
            .borrow()
            .is_empty()
    {
        // Step 1.1.
        let no_of_excess_entries_before =
                self.resource_timing_secondary_entries.borrow().len();
    
[...]
```

Reply:
Cool! Should I document all the methods I wrote?

Reply:
Yes, please.